pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -261,31 +261,38 @@ func getClusterGroupSourceRef(cg *crdv1alpha3.ClusterGroup) *controlplane.GroupR
 	}
 }
 
-func (c *NetworkPolicyController) triggerParentGroupSync(grp string) {
+func (c *NetworkPolicyController) triggerParentGroupUpdates(grp string) {
 	// TODO: if the max supported group nesting level increases, a Group having children
 	//  will no longer be a valid indication that it cannot have parents.
-	parentGroupObjs, err := c.internalGroupStore.GetByIndex(store.ChildGroupIndex, grp)
-	if err != nil {
-		klog.Errorf("Error retrieving parents of ClusterGroup %s: %v", grp, err)
-		return
-	}
+	parentGroupObjs, _ := c.internalGroupStore.GetByIndex(store.ChildGroupIndex, grp)
 	for _, p := range parentGroupObjs {
 		parentGrp := p.(*antreatypes.Group)
 		c.enqueueInternalGroup(parentGrp.SourceReference.ToGroupName())
 	}
 }
 
+// triggerDerivedGroupUpdates triggers processing of AppliedToGroup and AddressGroup derived from the provided group.
+func (c *NetworkPolicyController) triggerDerivedGroupUpdates(grp string) {
```

Comment:
This replaces the previous group enqueue operations in `reprocessCNP`, which blindly enqueues all AppliedToGroups and AddressGroups.